scripts/generator.go

```
@@ -0,0 +1,153 @@
+	interval   = flag.Duration("interval", 1*time.Second, "default duration to generate strings")
+	goroutines = flag.Int("goroutines", 2, "default go routines count")
+	address    = flag.String("address", "0.0.0.0", "addresses for zookeeper")
+)
+
+var mu = &sync.Mutex{}
+
+func main() {
+	flag.Parse()
+
+	if *goroutines > *element {
+		fmt.Printf("%v", fmt.Errorf("err: goroutines should be less than element number"))
+		return
+	}
+
+	addressArray := strings.Split(*address, ",")
+
+	c, _, err := zk.Connect(addressArray, time.Second) //*10)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// Before checking the zookeeper Data structure, clean all znodes from directory
+	if err := cleanUp(c); err != nil {
+		log.Fatal(err.Error())
+	}
+
+	sign := make(chan os.Signal)
+	signal.Notify(sign, os.Interrupt, syscall.SIGTERM)
+
+	closeChan := make(chan struct{}, 0)
+	go handleCtrlC(sign, closeChan)
+
+	go produce(*interval, *element, closeChan)
+
+	var wg sync.WaitGroup
+	wg.Add(*goroutines)
+	// multiple go routine to process created keys in zookeper data model
+	for i := 0; i < *goroutines; i++ {
+		go consume(c, &wg)
+	}
+
+	<-done
+	wg.Wait()
+
+	if producer == consumer {
+		fmt.Println(fmt.Sprintf("producer:%d and consumer:%d checking is finihed as successfully", producer, consumer))
+	}
+
+	os.Exit(0)
+}
+
+var (
+	producer, consumer int
+)
+
+func produce(x time.Duration, n int, c chan struct{}) {
+	defer func() {
+		close(msgs)
+		done <- true
+	}()
+
+	for i := 0; i < n; i++ {
+		select {
+		case <-c:
+			return
+		case <-time.After(x):
+			str := fmt.Sprintf("/key-%d", i)
+			msgs <- str
+			// we dont need to lock process coz there is single process
+			producer++
+		}
+	}
+}
+
+func consume(c *zk.Conn, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	for {
+		msg, ok := <-msgs
+		if !ok {
+			return
+		}
+		_, err := c.Create(msg, nil, 0, zk.WorldACL(zk.PermAll))
+		if err != nil {
+			if err != zk.ErrNodeExists {
```

yeah :) im gonna ignore this err here